pallets/treasury-buyout-extension/src/tests.rs

```diff
@@ -0,0 +1,642 @@
+			get_free_balance(GetNativeCurrencyId::get(), &TreasuryAccount::get());
+
+		assert_eq!(final_user_dot_balance, initial_user_dot_balance - exchange_amount);
+		assert_eq!(final_treasury_dot_balance, initial_treasury_dot_balance + exchange_amount);
+
+		assert_eq!(final_user_native_balance, initial_user_native_balance + buyout_amount);
+		assert_eq!(final_treasury_native_balance, initial_treasury_native_balance - buyout_amount);
+
+		// Verify Buyout event was emitted
+		assert!(System::events().iter().any(|record| matches!(
+			record.event,
+			TestEvent::TreasuryBuyoutExtension(crate::Event::Buyout { who, buyout_amount: amount, .. })
+			if who == user && amount == buyout_amount
+		)));
+	});
+}
+
+#[test]
+fn root_update_buyout_amount_limit_succeeds() {
+	run_test(|| {
+		let buyout_amount_limit = 200 * UNIT;
+		assert_ok!(crate::Pallet::<Test>::update_buyout_limit(
+			RuntimeOrigin::root(),
+			Some(buyout_amount_limit.into()),
+		));
+
+		assert_eq!(BuyoutLimit::<Test>::get(), buyout_amount_limit.into());
+
+		let buyout_amount_limit = None;
+		assert_ok!(crate::Pallet::<Test>::update_buyout_limit(
+			RuntimeOrigin::root(),
+			buyout_amount_limit,
+		));
+
+		assert_eq!(BuyoutLimit::<Test>::get(), buyout_amount_limit);
+	});
+}
+
+#[test]
+fn user_update_buyout_amount_limit_fails() {
+	run_test(|| {
+		let user = USER;
+
+		let buyout_amount_limit = 200 * UNIT;
+		assert_noop!(
+			crate::Pallet::<Test>::update_buyout_limit(
+				RuntimeOrigin::signed(user),
+				Some(buyout_amount_limit.into()),
+			),
+			BadOrigin
+		);
+	});
+}
+
+#[test]
+fn root_update_allowed_currencies_succeeds() {
+	run_test(|| {
+		let dot_currency_id = RelayChainCurrencyId::get();
+
+		// Since dot is already added to allowed currencies in the genesis, provide some other allowed currencies that will overwrite the storage
+		let allowed_currencies = vec![1u64, 2u64, 3u64];
+
+		assert_ok!(crate::Pallet::<Test>::update_allowed_assets(
+			RuntimeOrigin::root(),
+			allowed_currencies
+		));
+
+		// Test user buyout after allowed currencies update
+		// It should fail because dot is not allowed for buyout
+		let user = USER;
+		let exchange_amount = 100 * UNIT;
+
+		assert_noop!(
+			crate::Pallet::<Test>::buyout(
+				RuntimeOrigin::signed(user),
+				dot_currency_id,
+				Amount::Exchange(exchange_amount),
+			),
+			Error::<Test>::WrongAssetToBuyout
+		);
+
+		// Add dot back to allowed currencies among some others
+		let allowed_currencies = vec![
+			0u64,
+			3u64,
+			// Duplicating the same currency should not fail
+			3u64,
+			2u64,
+			dot_currency_id,
+			6u64,
+		];
+
+		assert_ok!(crate::Pallet::<Test>::update_allowed_assets(
+			RuntimeOrigin::root(),
+			allowed_currencies
+		));
+
+		// Test user buyout after allowed currencies update
+		// It should succeed because dot is now allowed for buyout
+		assert_ok!(crate::Pallet::<Test>::buyout(
+			RuntimeOrigin::signed(user),
+			dot_currency_id,
+			Amount::Exchange(exchange_amount),
+		));
+	});
```

Comment:
Perhaps we could compare with the event (or storage directly, probably faster) to ensure that the allowed currencies were indeed modified to the expected `allowed_currencies` vec, WDYT?

Reply:
Done 👍🏼